src/tonomyBankless/pages/tonomySwap.tsx

```diff
@@ -215,7 +215,14 @@ export default function Swap() {
       try {
         const direction: "tonomy" | "base" =
           currentDirection === SwapDirection.TONOMY_TO_BASE ? "base" : "tonomy";
-        await appUser.swapToken(new Decimal(toAmount), proof, direction);
+          if(direction === "base") {
+            const swapId = 'swap_' + Date.now() + '_' + Math.random().toString(36).substr(2, 9);
```

Comment:
better to use a cryptographically secure random number. SDK has a function to do this